### PR TITLE
add rescue InvalidURIError in mastodon_url validation

### LINKED ISSUE #6142: Invalid Mastodon URL Should Not Break Profile Update
Currently when a user tries to save an invalid Mastodon URL the save will cause a 500 error when we try to validate the URL when validating the user. 
Honeybadger: https://app.honeybadger.io/fault/66984/a02c794a66b30759ef45f2bfb5f09132
Error:
```
URI::InvalidURIError: bad URI(is not URI?): "https://mstdn.fr/@mickge\""
[PROJECT_ROOT]/vendor/ruby-2.6.5/lib/ruby/2.6.0/uri/rfc3986_parser.rb:67 :in `split`
65         ]
66       else
67         raise InvalidURIError, "bad URI(is not URI?): #{uri.inspect}"
68       end
69     end
[PROJECT_ROOT]/vendor/ruby-2.6.5/lib/ruby/2.6.0/uri/rfc3986_parser.rb:73 :in `parse`
[PROJECT_ROOT]/vendor/ruby-2.6.5/lib/ruby/2.6.0/uri/common.rb:234 :in `parse`
[PROJECT_ROOT]/app/models/user.rb:609 :in `validate_mastodon_url`
[PROJECT_ROOT]/vendor/ruby-2.6.5/lib/ruby/2.6.0/monitor.rb:235 :in `mon_synchronize`
[PROJECT_ROOT]/app/controllers/users_controller.rb:22 :in `update`
```
User model code raising the error:
```ruby
  def validate_mastodon_url
    return if mastodon_url.blank?

    uri = URI.parse(mastodon_url)
    return if uri.host&.in?(Constants::ALLOWED_MASTODON_INSTANCES)

    errors.add(:mastodon_url, "is not an allowed Mastodon instance")
  end
```

Rather than causing a 500 here we should catch the error and add the error message to the `mastodon_url` field. Possibly something like this.
```ruby
  def validate_mastodon_url
    return if mastodon_url.blank?

    uri = URI.parse(mastodon_url)
    return if uri.host&.in?(Constants::ALLOWED_MASTODON_INSTANCES)

    errors.add(:mastodon_url, "is not an allowed Mastodon instance")
  rescue URI::InvalidURIError => e
    errors.add(:mastodon_url, "url is invalid")
  end
```
Make sure to add a test for this as well! Thank you!

### PR DESCRIPTION
**Pull request by gonzar11**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
I've also removed mastodon_url validation from
```ruby
validates :website_url, :employer_url,
            url: { allow_blank: true, no_local: true, schemes: %w[https http] }
```
because we've already validated in `validate  :validate_mastodon_url`.

## Related Tickets & Documents
Fixes #6142

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -187,6 +187,11 @@ def user_from_authorization_service(service_name, signed_in_resource, cta_varian
         user.mastodon_url = "mastodon.social/@test"
         expect(user).not_to be_valid
       end
+
+      it "does not raise InvalidURIError when url is badly formatted" do
+        user.mastodon_url = "ben .com"
+        expect { user.validate }.not_to raise_error(URI::InvalidURIError)
```
Comment (citizen428) [inline spec/models/user_spec.rb]: Maybe we can test for the presence of the error instead of just testing that we don't raise an exception, WDYT?

Comment (gonzar11) [reply spec/models/user_spec.rb]: It makes sense for me. I will change it. Thanks!

**Comment by rhymes** [comment]
Hi @gonzar11, I think you used a different identity to post the latest commit, thus the CLA is asking for you to sign again

**Comment by gonzar11** [comment]
> Hi @gonzar11, I think you used a different identity to post the latest commit, thus the CLA is asking for you to sign again

Hi @rhymes , yes, I'm sorry, I got confused. I've just amended my commit. Is it ok now?

**Pull request merged**